### ISSUE TITLE
Migrating from wit.ai to rasa_nlu

### ISSUE THREAD

**Issue by Shravan40** [golastmile/rasa_nlu#153]
Hi there,

I have build a textParser with the help of wit.ai. But i would like to move it to my local machine for various reason (Not important here).

I have downloaded my project `SMSParser.zp` and extracted it. I followed the instruction given in the `README.md` file and server is running for for default set of data.

When i change the file `demo-rasa.son` with my `expressions.json` no changes have been observed. It only find the greeting intent. Please let me know, what am i doing wrong here ?

**Comment by amn41** [comment]
have you run `rasa_nlu.train` ?

**Comment by Shravan40** [comment]
```
┌─[shravan40@ChangeStreet] - [~/rasa_nlu] - [Mon Feb 13, 16:43]
└─[$] <git:(master)> python -m rasa_nlu.train -c config_mitie.json 
/usr/local/lib/python2.7/dist-packages/rasa_nlu-0.6_beta-py2.7.egg/rasa_nlu/__init__.py:1: UserWarning: Module rasa_nlu was already imported from /usr/local/lib/python2.7/dist-packages/rasa_nlu-0.6_beta-py2.7.egg/rasa_nlu/__init__.pyc, but /home/shravan40/rasa_nlu is being added to sys.path
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "build/bdist.linux-x86_64/egg/rasa_nlu/train.py", line 65, in <module>
  File "build/bdist.linux-x86_64/egg/rasa_nlu/train.py", line 59, in do_train
  File "build/bdist.linux-x86_64/egg/rasa_nlu/trainers/mitie_trainer.py", line 26, in train
  File "build/bdist.linux-x86_64/egg/rasa_nlu/trainers/mitie_trainer.py", line 62, in train_intent_classifier
  File "/usr/local/lib/python2.7/dist-packages/mitie/mitie.py", line 765, in __init__
    raise Exception("Unable to create text_categorizer_trainer based on " + filename)
Exception: Unable to create text_categorizer_trainer based on ./data/total_word_feature_extractor.dat
┌─[shravan40@ChangeStreet] - [~/rasa_nlu] - [Mon Feb 13, 16:43]
└─[$] <git:(master)> 
```

**Comment by Shravan40** [comment]
Hey @amn41 

Here is the content of my `config.json`
```
{
  "backend": "mitie",
  "mitie_file": "/home/shravan40/Downloads/MITIE-models/english/total_word_feature_extractor.dat",
  "path" : "./",
  "data" : "/home/shravan40/Downloads/SMSParser/expressions.json"
}
```

When i run `python -m rasa_nlu.train -c config.json`. After a few minutes, rasa NLU will finished training but no directory of  type `model_YYYYMMDD-HHMMSS` has been created. 

But in logs i have found that `Entities must span whole tokens` quite often. Here is the last few lines of log
```
Training to recognize 1 categories: 'None'
Train classifier
extracting text features
now do training
num training samples: 504
C: 200   f-score: 1
C: 400   f-score: 1
C: 300   f-score: 1
C: 100   f-score: 1
C: 0.01   f-score: 1
C: 50.005   f-score: 1
C: 25.0075   f-score: 1
C: 12.5088   f-score: 1
C: 6.25938   f-score: 1
C: 3.13469   f-score: 1
C: 1.57234   f-score: 1
C: 0.791172   f-score: 1
C: 0.400586   f-score: 1
best C: 0.791172
test on train: 
504 

overall accuracy: 1
Training time: 1 seconds.
df.number_of_classes(): 1

Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "build/bdist.linux-x86_64/egg/rasa_nlu/train.py", line 65, in <module>
  File "build/bdist.linux-x86_64/egg/rasa_nlu/train.py", line 59, in do_train
  File "build/bdist.linux-x86_64/egg/rasa_nlu/trainers/mitie_trainer.py", line 30, in train
  File "build/bdist.linux-x86_64/egg/rasa_nlu/trainers/mitie_trainer.py", line 53, in train_entity_extractor
  File "build/bdist.linux-x86_64/egg/rasa_nlu/trainers/mitie_trainer.py", line 39, in find_entity
ValueError: invalid entity {u'start': 87, u'end': 99, u'role': u'merchantName', u'value': u'PayTm Mobil.', u'entity': u'message_body'} in example Dear Customer, You have made a Debit Card purchase of INR 1,389.00 on 28 Oct. Info.IIN*PayTm Mobil. Your Net Available Balance is INR  3,625.58.: entities must span whole tokens
```
And when is run `python -m rasa_nlu.server -c config.json` it says 

```
/usr/local/lib/python2.7/dist-packages/rasa_nlu-0.6_beta-py2.7.egg/rasa_nlu/__init__.py:1: UserWarning: Module rasa_nlu was already imported from /usr/local/lib/python2.7/dist-packages/rasa_nlu-0.6_beta-py2.7.egg/rasa_nlu/__init__.pyc, but /home/shravan40/rasa_nlu is being added to sys.path
{
    "log_level": 20, 
    "language": "en", 
    "mitie_file": "/home/shravan40/Downloads/MITIE-models/english/total_word_feature_extractor.dat", 
    "emulate": null, 
    "config": "config.json", 
    "write": "/home/shravan40/rasa_nlu/rasa_nlu_log.json", 
    "token": null, 
    "server_model_dir": null, 
    "path": "./", 
    "log_file": null, 
    "data": "/home/shravan40/Downloads/SMSParser/expressions.json", 
    "port": 5000, 
    "backend": "mitie"
}
WARNING:root:backend 'mitie' specified in config, but no model directory is configured. Using 'hello-goodby' backend instead!
INFO:root:using default hello-goodby backend
INFO:root:Started http server on port 5000
```

But it's not parsing any kind of data.

**Comment by amn41** [comment]
Yep - you should manually split the `Info.IIN*PayTm` token before putting into rasa. 

The problem is that the tokenizer doesn't split on the `*` character, which means that `PayTm` is just the 2nd half of a single `Info.IIN*PayTm` token. You can't have 'half of a token' as part of an entity.
